plugins/aws/timestreamwrite/timestreamDatabaseEncrypted.spec.js

```diff
@@ -0,0 +1,152 @@
+            "Enabled": true,
+            "Description": "Default master key that protects my Glue data when no other key is defined",
+            "KeyUsage": "ENCRYPT_DECRYPT",
+            "KeyState": "Enabled",
+            "Origin": "AWS_KMS",
+            "KeyManager": "CUSTOMER",
+            "CustomerMasterKeySpec": "SYMMETRIC_DEFAULT",
+            "EncryptionAlgorithms": [
+                "SYMMETRIC_DEFAULT"
+            ]
+        }
+    },
+    {
+        "KeyMetadata": {
+            "AWSAccountId": "000011112222",
+            "KeyId": "2cff2321-73c6-4bac-95eb-bc9633d3e8a9",
+            "Arn": "arn:aws:kms:us-east-1:000011112222:key/c4750c1a-72e5-4d16-bc72-0e7b559e0250",
+            "CreationDate": "2020-12-15T01:16:53.045000+05:00",
+            "Enabled": true,
+            "Description": "Default master key that protects my Glue data when no other key is defined",
+            "KeyUsage": "ENCRYPT_DECRYPT",
+            "KeyState": "Enabled",
+            "Origin": "AWS_KMS",
+            "KeyManager": "AWS",
+            "CustomerMasterKeySpec": "SYMMETRIC_DEFAULT",
+            "EncryptionAlgorithms": [
+                "SYMMETRIC_DEFAULT"
+            ]
+        }
+    }
+];
+
+const createCache = (databases, keys, describeKey, databasesErr, keysErr, describeKeyErr) => {
+    var keyId = (databases && databases.length && databases[0].KmsKeyId) ? databases[0].KmsKeyId.split('/')[1] : null;
+    return {
+        timestreamwrite: {
+            listDatabases: {
+                'us-east-1': {
+                    err: databasesErr,
+                    data: databases
+                },
+            },
+        },
+        kms: {
+            listKeys: {
+                'us-east-1': {
+                    data: keys,
+                    err: keysErr
+                }
+            },
+            describeKey: {
+                'us-east-1': {
+                    [keyId]: {
+                        err: describeKeyErr,
+                        data: describeKey
+                    },
+                },
+            },
+        },
+    };
+};
+
+
+describe('timestreamDatabaseEncrypted', function () {
+    describe('run', function () {
+        it('should PASS if Timestream database is encrypted with desired encryption level', function (done) {
+            const cache = createCache([listDatabases[1]], listKeys, describeKey[0]);
+            timestreamDatabaseEncrypted.run(cache, { timestream_databases_desired_encryption_level: 'awscmk' }, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Timestream database is encrypted with awscmk');
+                done();
+            });
+        });
+
+
+        it('should FAIL if Timestream database is not encrypted with desired encyption level', function (done) {
+            const cache = createCache([listDatabases[0]], listKeys, describeKey[1]);
```

Comment:
Here the keys are different in listDatabases[0] and describeKey[1]. Please check.